Sources/State/LoadingMediaState.swift

```diff
@@ -22,36 +22,56 @@ public final class LoadingMediaState: PlayerState {
     private let url: URL?
     private let lastKnownPosition: CMTime?
     private var itemStatusObserving: ItemStatusObservingService?
+    private var interruptionAudioService: InterruptionAudioService
 
     // MARK: - Init
     
-    public init(context: PlayerContext, itemUrl: URL, shouldPlaying: Bool, lastPosition: CMTime?) {
+    public init(context: PlayerContext,
+                itemUrl: URL,
+                shouldPlaying: Bool,
+                lastPosition: CMTime?,
+                interruptionAudioService: InterruptionAudioService = InterruptionAudioService()) {
         LoggerInHouse.instance.log(message: "Init", event: .debug)
         self.context = context
         self.shouldPlaying = shouldPlaying
         self.media = nil
         self.url = itemUrl
         self.lastKnownPosition = lastPosition
+        self.interruptionAudioService = interruptionAudioService
 
+        context.audioSessionType.active { _ in }
+        
+        setupInterruptionCallback()
         startBgTask(context: context)
         createReplaceItem(url: itemUrl)
     }
 
-    public init(context: PlayerContext, media: PlayerMedia, shouldPlaying: Bool) {
+    public init(context: PlayerContext,
+                media: PlayerMedia,
+                shouldPlaying: Bool,
+                interruptionAudioService: InterruptionAudioService = InterruptionAudioService()) {
         LoggerInHouse.instance.log(message: "Init", event: .debug)
         self.context = context
         self.shouldPlaying = shouldPlaying
         self.media = media
         self.url = nil
         self.lastKnownPosition = nil
-
+        self.interruptionAudioService = interruptionAudioService
+        
+        context.audioSessionType.active { _ in }
```

Comment:
Maybe it would be better to change active function to take an optional completion
```
static public func active(completion: ((Bool) -> Void)?)
```
See https://oleb.net/blog/2016/10/optional-non-escaping-closures/ for escaping attribute

Reply:
And at the same time, can you rename it to 'activate'?

Reply:
Actually, no need anymore about the completion at this time